### PR TITLE
Doc: split "advanced topics" page

### PR DESCRIPTION
**Pull request by emillon**
This is the exact same as #7232 but for our other collection page: "advanced
topics".

See #7325 for the general plan of how to get to a better place in terms of
documentation.

### CODE REVIEW

File: doc/advanced/findlib-integration.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+- ``mt_posix``: forces the use of POSIX threads rather than VM
+  threads. VM threads are deprecated and will soon be obsolete.
+
+- ``ppx_driver``: when a library acts differently depending on whether
+  it's linked as part of a driver or meant to add a ``-ppx`` argument
+  to the compiler, choose the former behavior.
+
+Note that Dune does not read *installed* ``META`` files for libraries
+distributed with the compiler (as these files are not installed by the compiler
+itself, but installed by `ocamlfind` and aren't always
```
Comment (christinerose) [inline doc/advanced/findlib-integration.rst:30]:
```suggestion
itself, but installed by ``ocamlfind`` and aren't always 
```
Should this be in monospace? If only italicised, it should be OCamlFind (I think, as we write OCamlFormat like this.)

File: doc/advanced/package-version.rst [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Package Version
+===============
+
+Dune determines a package's version by looking at the ``version``
+field in the :ref:`package stanza <package>`. If the version field isn't 
+set, it looks at the toplevel ``version`` field in the
+``dune-project`` field. If neither are set, Dune assumes that we are in
+development mode and reads the version from the VCS if any. The way it
```
Comment (christinerose) [inline doc/advanced/package-version.rst:8]:
```suggestion
development mode and reads the version from the VCS, if any. The way it
```

File: doc/advanced/package-version.rst [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+obtains the version from the VCS in described in :ref:`the build-info
+section <build-info>`.
+
+When installing the files of a package on the system, Dune
```
Comment (christinerose) [inline doc/advanced/package-version.rst:12]:
```suggestion
When installing package files on the system, Dune
```

File: doc/advanced/variables-artifacts.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+expected and also inside :doc:`../concepts/actions`. When used inside
+:doc:`../concepts/actions`, they implicitly declare a dependency on the
+corresponding artifact.
+
+The variables have the form ``%{<ext>:<path>}``, where ``<path>`` is
+interpreted relative to the current directory:
+
+- ``cmo:<path>``, ``cmx:<path>``, and ``cmi:<path>`` expand to the corresponding
+  artifact's path for the module specified by ``<path>``. The basename of
+  ``<path>`` should be the name of a module as specified in a ``(modules)``
```
Comment (christinerose) [inline doc/advanced/variables-artifacts.rst:21]:
```suggestion
  ``<path>`` should be the module's name as specified in a ``(modules)``
```

**Review by christinerose — APPROVED** [review]
Excellent! I only have a few minor suggestions.

File: doc/advanced/findlib-integration.rst (hunk 1 above)
Comment (emillon) [reply doc/advanced/findlib-integration.rst:30]: yes, that should monospace. OCamlFormat is the exception I think, the author of Findlib uses "Findlib" and "ocamlfind" on http://projects.camlcity.org/projects/findlib.html

**Pull request merged**